tidy3d/components/simulation.py

```diff
@@ -1329,7 +1329,12 @@ def snap_to_grid(geom: Geometry, axis: Axis) -> Geometry:
         # Convert lumped elements into structures
         lumped_structures = []
         for lumped_element in self.lumped_elements:
-            lumped_structures.append(lumped_element.to_structure(self.grid))
+            if isinstance(lumped_element, LinearLumpedElement):
+                PEC_connection = lumped_element.to_PEC_connections(self.grid)
+                if PEC_connection is not None:
+                    lumped_structures.append(PEC_connection)
+            lumped_structure = lumped_element.to_structure(self.grid)
+            lumped_structures.append(lumped_structure)
```

Comment:
Minor: probably better to shift the logic to lumped_element method, as this part doesn't depend on Simulation class. E.g. you can add `to_structures` method to LumpedElement.